### ISSUE TITLE
sms77io not selectable

### ISSUE THREAD

**Issue by itverwaltung** [nextcloud/twofactor_gateway#302]
In version 15 smsio is supported as a twofactor gateway provider.

www-data@svrhnextcloud:/var/www/nextcloud$ php occ twofactorauth:gateway:configure sms
Please choose a SMS provider (websms, playsms, clockworksms, puzzelsms, ecallsms, voipms, huawei_e3531, spryng): 

sms77io is not listed... How can i use this provider?

www-data@svrhnextcloud:/var/www/nextcloud$ php occ app:list | grep twofactor_gateway

  - twofactor_gateway: 0.15.0

So version 15 is installed ...

Thank you for any advice..

Alex

**Comment by ChristophWurst** [comment]
Hi Alex,

enter 'sms77' ;)

It's just missing in the help text at https://github.com/nextcloud/twofactor_gateway/blob/c7a23e6eb77fa5a44465d2f870a8e418e687b1ce/lib/Command/Configure.php#L108. Mind submitting a pull request to add it? That would be awesome!

Cheers and thanks for the feedback.

**Comment by itverwaltung** [comment]
Hi Christoph,

i did as you said, but no luck:

www-data@svrhnextcloud:/var/www/nextcloud$ php occ twofactorauth:gateway:configure sms
Please choose a SMS provider (websms, playsms, clockworksms, puzzelsms, ecallsms, voipms, huawei_e3531, spryng): sms77

In ProviderFactory.php line 59:

  Provider <sms77> does not exist


twofactorauth:gateway:configure <gateway>

www-data@svrhnextcloud:/var/www/nextcloud$


Any further hints?
Thank you!

Kind regards

Alex

**Comment by ChristophWurst** [comment]
Correct. There is another bug :see_no_evil: 

Please submit another PR that changes https://github.com/nextcloud/twofactor_gateway/blob/c7a23e6eb77fa5a44465d2f870a8e418e687b1ce/lib/Command/Configure.php#L239 to 'sms77io'. Then it should work.

(ref https://github.com/nextcloud/twofactor_gateway/blob/c7a23e6eb77fa5a44465d2f870a8e418e687b1ce/lib/Service/Gateway/SMS/Provider/Sms77Io.php#L33)

**Comment by itverwaltung** [comment]
I tried to make a fork, a branch, edited the file, commited, and made a pull request. First time, have mercy ...

**Comment by ChristophWurst** [comment]
No worries. We all started there :)

The DCO check failed. You might have to rebase the commits locally. Let me knew if you need help. You'll also find useful information at https://help.github.com/en/github

**Comment by nursoda** [comment]
Christoph, seriously? We are users (or admins) but not all of us are devs. For us it's hard enough to post decent bug reports. But requesting multiple PR from an oviously non-dev where it would have been two tiny code changes you already looked up seems insane to me.

**Comment by ChristophWurst** [comment]
Wow, wait a sec. This app is a community effort. There is no obligation for anyone to maintain this project.

I've been very permissive towards the many community PRs that add support for edge case gateways. There is no way for me to test them, nor maintain it in a way that I would know if they work.

If you need enterprise support for this app you can check out https://nextcloud.com/enterprise/. But otherwise this is and will remain a community app.

Doing the maintenance of managing PRs, triaging issues and helping with problems is the most I can do. The change is minimal. But it takes me as much time as it would take you.

**Comment by ChristophWurst** [comment]
And you don't have to code yourself if you're not a programmer. Just ask or hire someone to do it if it's important.

**Comment by nursoda** [comment]
Fair enough. Thanks for committing. I confirm that sms77.io works fine :-)

**Comment by cl77** [comment]
The `sms77io` provider has been selectable in the SMS provider picker for a long time (it shows up as option `sms77.io` in `occ twofactorauth:gateway:configure sms`). Closing as fixed.

For anyone landing here: the driver was modernized in #1022 (renamed to **seven**, switched to the current `gateway.seven.io` endpoint, with proper per-recipient delivery validation). The settings id stays `sms77io`, so existing configurations keep working.

**Comment by nursoda** [comment]
Shouldn't this issue be closed then?

**Comment by cl77** [comment]
Yes, can be closed.
